mobile/geth.go

```diff
@@ -185,7 +217,56 @@ func NewNode(datadir string, config *NodeConfig) (stack *Node, _ error) {
 			return nil, fmt.Errorf("whisper init: %v", err)
 		}
 	}
-	return &Node{rawStack}, nil
+	if config.PssEnabled && ks != nil {
+		log.Debug("pss enabled")
+		bzzSvc := func(ctx *node.ServiceContext) (node.Service, error) {
+			log.Debug("keystore", "ks", ks)
+			var a accounts.Account
```

Comment:
can you explain why all this juggling is needed . Is this not taken care of by the `swarm.New` call?